parseAndPopulate/modules.py

```diff
@@ -61,111 +73,143 @@ def __init__(self, name: str, path: str, schemas: dict, dir_paths: DirPaths,
         except yangParser.ParseException:
-            LOGGER.exception('Missing yang file {}'.format(self._path))
+            self.logger.exception(f'Missing yang file {self._path}')
             raise
-        self.implementations: t.List[Implementation] = []
-        self._parse_all(name, yang_modules, additional_info)
 
-    def _parse_all(self, name: str, yang_modules: dict, additional_info: t.Optional[t.Dict[str, str]] = None):
-        if not additional_info:
-            additional_info = {}
+    def _parse_all(self, name: str, yang_modules: dict, additional_info: t.Optional[dict[str, str]]):
+        additional_info = additional_info or {}
         self.author_email = additional_info.get('author-email')
         self.maturity_level = additional_info.get('maturity-level')
         self.reference = additional_info.get('reference')
         self.document_name = additional_info.get('document-name')
-        self.module_classification = additional_info.get(
-            'module-classification', 'unknown')
+        self.module_classification = additional_info.get('module-classification', 'unknown')
         generated_from = additional_info.get('generated-from')
         organization = additional_info.get('organization')
         self.compilation_status = 'unknown'
         self.compilation_result = None
         self.ietf_wg = None
 
         self.name: str = self._parsed_yang.arg or name
-        revision_resolver = RevisionResolver(self._parsed_yang, LOGGER)
+        revision_resolver = RevisionResolver(self._parsed_yang, self.logger)
         self.revision = revision_resolver.resolve()
-        name_revision = '{}@{}'.format(self.name, self.revision)
+        name_revision = f'{self.name}@{self.revision}'
 
         belongs_to_resolver = BasicResolver(self._parsed_yang, 'belongs_to')
         self.belongs_to = belongs_to_resolver.resolve()
 
-        namespace_resolver = NamespaceResolver(
-            self._parsed_yang, LOGGER, name_revision, self.belongs_to)
+        namespace_resolver = NamespaceResolver(self._parsed_yang, self.logger, name_revision, self.belongs_to)
         self.namespace = namespace_resolver.resolve()
 
-        organization_resolver = OrganizationResolver(
-            self._parsed_yang, LOGGER, self.namespace)
+        organization_resolver = OrganizationResolver(self._parsed_yang, self.logger, self.namespace)
         self.organization = organization or organization_resolver.resolve()
 
-        module_type_resolver = ModuleTypeResolver(self._parsed_yang, LOGGER)
+        module_type_resolver = ModuleTypeResolver(self._parsed_yang, self.logger)
         self.module_type = module_type_resolver.resolve()
 
-        key = '{}@{}/{}'.format(self.name, self.revision, self.organization)
+        key = f'{self.name}@{self.revision}/{self.organization}'
         if key in yang_modules:
             return
+        if self.can_be_already_stored_in_db and (module_data := self._redis_connection.get_module(key)) != '{}':
```

Comment:
Added ```... and self._redis_connection and ...``` to not make a database call because ```self._redis_connection``` is always ```True``` for now (as it's passed from ```groupings.py``` every time)